apis/core.oam.dev/v1alpha1/garbagecollect_policy_types.go

```diff
@@ -73,6 +76,16 @@ const (
 	GarbageCollectStrategyOnAppUpdate GarbageCollectStrategy = "onAppUpdate"
 )
 
+// GarbageCollectPropagation the deletion propagation setting similar to metav1.DeletionPropagation
+type GarbageCollectPropagation string
+
+const (
+	// GarbageCollectPropagationOrphan orphan child resources while deleting target resources
+	GarbageCollectPropagationOrphan = "orphan"
+	// GarbageCollectPropagationBackground delete child resources in background while deleting target resources
+	GarbageCollectPropagationBackground = "background"
```

Comment:
How about `cascading`? I think backgroud is kinda confusing since we don't allow `foreground` GC option.

Reply:
Actually, "foreground" is okay for client-go but we should not allow it here. It will block the controller and should only be a CLI case.

Reply:
I also think `cascading` as a good choice as well.

Reply:
Renamed to `cascading`.

Reply:
The Type still named as Propagation to aligned with the controller-runtime setting.

Reply:
I think about "cascading" as a good choice because that's how kubectl use for flags. But I later confirmed from the bare API parameter and see the PropagationPolicy used for DELETE api.
![image](https://github.com/kubevela/kubevela/assets/14019297/498d5e44-e43d-480b-a3fd-07c7fe1735c7)

Reply:
Great, I renamed the value of "background" to "cascading".